Workaround for [0]. In bemenu monitor indices start at 1, while
they start at zero with dmenu. Explicitly passing -m is neccessary
in the first place, as X11 is not able to tell the current monitor
if no window is focused on it. To reproduce, switch to an empty
workspace on your second screen, switch to the first screen, switch back
to the second screen and start dmenu/bemenu, the menu will appear on the
first screen [1].

[0]: https://github.com/Cloudef/bemenu/issues/164
[1]: https://lists.suckless.org/dev/1208/12167.html

diff -upr a/dwm.c b/dwm.c
--- a/dwm.c	2020-06-18 16:17:21.000000000 +0200
+++ b/dwm.c	2021-04-04 14:38:06.908113987 +0200
@@ -1694,7 +1694,7 @@ void
 spawn(const Arg *arg)
 {
 	if (arg->v == dmenucmd)
-		dmenumon[0] = '0' + selmon->num;
+		dmenumon[0] = '0' + selmon->num + 1;
 	if (fork() == 0) {
 		if (dpy)
 			close(ConnectionNumber(dpy));
